# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.h
# Commit: 9da00c1364a5
# Full Hash: 9da00c1364a5b2bb35ca16205dc927ae8186a984
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 1
# Description:
#   Bug 1351231 - Support conversation between InternalRequest and IPCInternalRequest. r=dom-worker-reviewers,jesup
#   
#   Depends on D138812
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138813
# ==============================================================================

diff -r 19b78046ffef -r 9da00c1364a5 dom/fetch/InternalRequest.h
--- a/dom/fetch/InternalRequest.h	Wed Jan 18 00:50:18 2023 +0000
+++ b/dom/fetch/InternalRequest.h	Wed Jan 18 00:50:19 2023 +0000
@@ -29,6 +29,7 @@
 namespace mozilla {
 
 namespace ipc {
+class PBackgroundChild;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -97,6 +98,9 @@
 
   explicit InternalRequest(const IPCInternalRequest& aIPCRequest);
 
+  void ToIPCInternalRequest(IPCInternalRequest* aIPCRequest,
+                            mozilla::ipc::PBackgroundChild* aManager);
+
   SafeRefPtr<InternalRequest> Clone();
 
   void GetMethod(nsCString& aMethod) const { aMethod.Assign(mMethod); }